### ISSUE TITLE
how do I define a "placeholder" module in which to inject a library dependency after the build is done?

### ISSUE THREAD

**Issue by UnsungHero97** [webpack/webpack#2082]
I have a dependency on a library that is generated by [emscripten](https://github.com/kripken/emscripten), and I'd like to bundle it alongside the library I'm building. However, I can't figure out how to configure webpack to do what I want, which is to bundle all of my modules / dependencies and uglify the result, except without touching the emscripten-generated library.

I want webpack to leave the emscripten-generated library alone, and the reason is, it looks like this: 

<img width="841" alt="screen shot 2016-02-20 at 7 37 36 pm" src="https://cloud.githubusercontent.com/assets/440299/13200718/894a85ba-d809-11e5-9c21-dd9b5e2d972c.png">

In other words, it is highly optimized `asm.js` JavaScript, compiled from C++, and it's already uglified; it doesn't need to be parsed (I know about `module.noParse`). Unfortunately, my current `webpack.config.js` bundles and uglifies everything, including the emscripten-generated library.

Is there a way to bundle and uglify everything, using a "placeholder" module for the emscripten-generated library, and then inserting it as the last step?

**Comment by sokra** [comment]
You should report this to UglifyJs. This should be solved in that library (as it's not a webpack-specific problem).

As workaround: you can put it into a on-demand-loaded chunk and exclude the whole chunk with the `exclude` option in the UglifyJsPlugin.

**Comment by UnsungHero97** [comment]
@sokra I can report to UglifyJs as well, but I think this is very much a webpack question. The core of what I'm asking is in regards to dependency injection. After all the loaders and plugins finish running, I want to have webpack inject the library untouched in it's proper module.

**Comment by sokra** [comment]
Uglify doesn't run on each module. It runs once on the final bundle with all modules concatentated.

**Comment by UnsungHero97** [comment]
I'm aware of that.

**Comment by UnsungHero97** [comment]
@sokra I'm trying to switch focus away from UglifyJs to webpack. I understand that UglifyJs runs against the bundle, not per module.

I'm just wondering if there's a way to configure webpack to use a "placeholder" module during build time in which to inject my dependency after all plugins have finished, as the last step in the build process?

**Comment by sokra** [comment]
> I'm just wondering if there's a way to configure webpack to use a "placeholder" module during build time in which to inject my dependency after all plugins have finished, as the last step in the build process?

Possible, but I don't have time for such a plugin.

**Comment by bebraw** [comment]
As discussed, this would need a plugin. Feel free to contribute.
